vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -613,6 +612,16 @@ def register_kv_caches(self, kv_caches: dict[str, torch.Tensor]):
         logger.debug("Done registering descs")
         self._registered_descs.append(descs)
 
+        if self._use_flashinfer:
+            # NOTE (NickLucche) When FlahsInfer is used, memory is registered
+            # with joint KV for each block. This minimizes the overhead in
+            # registerMem allowing faster descs queries. In order to be able to
+            # split on kv_heads dim as required by heterogeneous TP, one must
+            # be able to index K/V separately. Hence the we double the number
+            # of 'virtual' regions here and double the descs below.
+            self.num_regions *= 2
+
+        block_len = self.get_backend_aware_block_len()
```

Comment:
Rare instance where I think the helper method actually makes it a little harder to understand what's going on.
```suggestion
            block_len = self.block_len // 2
```
Also there's a landmine here that should be managed -- someone may try to use `self.block_len` instead of `block_len` and run into issues here. Does it make sense to halve `self.block_len` in the flashinfer case instead?